logstash-core/lib/logstash/util/dead_letter_queue_factory.rb

```diff
@@ -0,0 +1,32 @@
+require 'logstash/environment'
+
+module LogStash; module Util
+  class DeadLetterQueueWriter
+
+    def initialize(inner_writer)
+      @inner_writer = inner_writer
+    end
+
+    def write(logstash_event, plugin_type, plugin_id, reason)
+      if @inner_writer && @inner_writer.is_open
+        @inner_writer.write_entry(logstash_event.to_java, plugin_type, plugin_id, reason)
+      end
+    end
+  end
+
+  class DeadLetterQueueFactory
```

Comment:
I am not sure about the naming of this `class`, It feels more like a lifecycle manager to me because it's also used to DLQ. Maybe it's me but a factory its more like give me something that I can work on.